src/middlewared/middlewared/plugins/reporting/iostat.py

```diff
@@ -0,0 +1,77 @@
+from collections import namedtuple
+import time
+
+DiskStat = namedtuple("DiskStat", ["rd_ios", "rd_bytes", "wr_ios", "wr_bytes", "tot_ticks"])
+
+
+class DiskStats:
+    def __init__(self):
+        self.prev = None
+        self.prev_time = None
+        self.current = None
+        self.current_time = None
+        self._read()
+
+    def _read(self):
+        with open("/proc/diskstats") as f:
+            lines = f.readlines()
+
+        self.current = {}
+        for line in lines:
+            line = line.split()
+
+            disk = line[2]
+
+            if disk.startswith("loop"):
+                continue
+
+            parent = disk
+            while parent and (parent[-1].isdigit() or parent[-1] == "p" and parent.startswith("nvme")):
+                parent = parent[:-1]
+            if parent in self.current:
+                continue
+
+            disk_stat = DiskStat(
+                rd_ios=int(line[3]),
+                rd_bytes=int(line[5]) * 512,
```

Comment:
Why 512?

Reply:
These are sectors which are always 512 bytes as kernel source says https://github.com/torvalds/linux/blob/master/include/linux/types.h#L120

Reply:
Thanks for enlightening me. Might be worth putting a comment explaining that since its not clear every device would be treat as such.